tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlockAwareSegmentInputStreamImpl.java

```diff
@@ -186,6 +275,10 @@ public void close() throws IOException {
             entriesByteBuf.forEach(buf -> buf.release());
             entriesByteBuf.clear();
         }
+        if (paddingBuf.refCnt() != 0) {
```

Comment:
this is very dangerous.
we are blindly decreasing the ref count
we should decrease the ref count only if we incremented it and the same number of times

Reply:
+1

Reply:
My bad. That is a test code and I push it accidentally. The root reason is the close method called twice. 
The close method will be triggered twice in the BlobStoreManagedLedgerOffloader#offload method. The stream resource used by the try-with block will be called the close. And through debug, writeBlobStore.uploadMultipartPart in the offload method also will trigger the close method.